packages/addon-dev/src/rollup-public-entrypoints.ts

```diff
@@ -16,14 +16,14 @@ export default function publicEntrypoints(args: {
   return {
     name: 'addon-modules',
     async buildStart() {
-      this.addWatchFile(args.srcDir);
-
       let matches = walkSync(args.srcDir, {
         globs: [...args.include, '**/*.hbs', '**/*.ts', '**/*.gts', '**/*.gjs'],
         ignore: args.exclude,
```

Comment:
Now it will take into account `exclude` option and exclude files from watch mode as well